scripts/install.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/sh
+
+REPO="https://github.com/software-mansion/cairo-profiler"
+BINARY_NAME="cairo-profiler"
+LOCAL_BIN="${HOME}/.local/bin"
+
+main () {
+  check_cmd curl
+  check_cmd tar
+
+  version=${1:-latest}
+  release_tag=$(curl -# --fail -Ls -H 'Accept: application/json' "${REPO}/releases/{$version}" | sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
+
+  if [ -z "$release_tag" ]; then
+    echo "No such version $version, please pass correct one (e.g. v0.1.0)"
+    exit 1
+  fi
+
+  download_and_extract_binary "$release_tag"
+
+  add_binary_to_path
+
+  echo "${BINARY_NAME} (${release_tag}) has been installed successfully."
+}
+
+check_cmd() {
+  if ! command -v "$1" >/dev/null 2>&1; then
+      echo "$1 is not available"
+      echo "Please install $1 and run the script again"
+      exit 1
+  fi
+}
+
+download_and_extract_binary() {
+  release_tag=$1
+
+  # Define the operating system and architecture
+  get_architecture
+  _arch="$RETVAL"
+
+  artifact_name=${BINARY_NAME}-${release_tag}-${_arch}
+
+  echo "Downloading and extracting ${artifact_name}..."
+  # Create a temporary directory
+  temp_dir=$(mktemp -d)
+
+  # Download and extract the archive
+  curl -L "${REPO}/releases/download/${release_tag}/${artifact_name}.tar.gz" | tar -xz -C "${temp_dir}"
+
+  # Move the binary to a LOCAL_BIN directory
+  mkdir -p "${LOCAL_BIN}"
+  mv "${temp_dir}/${artifact_name}/bin/${BINARY_NAME}" "${LOCAL_BIN}"
+
+  # Clean up temporary files
+  rm -rf "${temp_dir}"
+}
+
+get_architecture() {
+  local _ostype _cputype _arch _clibtype
```

Comment:
nit: `local` is kind of a bash-ism, and is not posix compatible in shell (sh). Generally this should work, but there migth be an edge case on some OSes/shells where it just won't, yielding an error.

context: https://www.shellcheck.net/wiki/SC3043